dgc/GreenCertificateVerifier.cs

```diff
@@ -17,28 +18,30 @@ public GreenCertificateVerifier(SecretariatService secretariatService)
         /// </summary>
         /// <param name="coseMessage">Cose message</param>
         /// <returns>(true, null) if valid, (false, reason) if not valid</returns>
-        public (bool, string) Verify(CWT cwt)
+        public async Task<Tuple<bool, string>> Verify(CWT cwt, DateTime? verifyClock = null)
         {
-            var publicKeys = _secretariatService.GetPublicKeys(cwt.CoseMessage.KID);
-            if (publicKeys.Any())
+            if (!verifyClock.HasValue) verifyClock = DateTime.Now;
+
+            var certificates = await _secretariatService.GetCertificate(cwt.CoseMessage.KID);
```

Comment:
`GetCertificates(...)` ?

Reply:
changed this to count>0 because using IReadOnlyList